eslint.config.mjs

```diff
@@ -0,0 +1,164 @@
+import ts from '@typescript-eslint/eslint-plugin'
+import tsParser from '@typescript-eslint/parser'
+import prettier from 'eslint-config-prettier'
+import react from 'eslint-plugin-react'
+import reactHooks from 'eslint-plugin-react-hooks'
+import testingLibrary from 'eslint-plugin-testing-library'
+import jest from 'eslint-plugin-jest'
+import globals from 'globals'
+
+export default [
+  'eslint:recommended',
+  // Ignored dirs
+  {
+    ignores: ['dist/**/*', 'compiled/**/*', 'bundle/**/*']
+  },
+  // Temporary ignored dirs - TODO: remove on rewrite
+  {
+    ignores: ['main/signers/**/*']
+  },
+  // All files
+  {
+    files: ['**/*.{js,mjs,ts,tsx}'],
+    languageOptions: {
+      ecmaVersion: 'latest',
+      globals: {
+        ...globals.es2021
+      }
+    },
+    rules: {
+      'no-unused-vars': [
+        'error',
+        {
+          args: 'after-used',
+          ignoreRestSiblings: true,
+          argsIgnorePattern: '^_',
+          destructuredArrayIgnorePattern: '^_'
+        }
+      ]
+    }
+  },
+  // Main process files and scripts
+  {
+    files: ['**/*.{js,mjs,ts}'],
+    ignores: ['app/**/*', 'resources/Components/**/*'],
+    languageOptions: {
+      globals: {
+        ...globals.node
+      }
+    }
+  },
+  // Renderer process files
+  {
+    files: [
+      'app/**/*.js',
+      'main/dapps/server/inject/*.js',
+      'resources/**/*.{js,ts,tsx}',
+      'test/app/**/*.js',
+      'test/resources/Components/**/*.js'
+    ],
+    languageOptions: {
+      globals: {
+        ...globals.browser,
+        process: true,
```

Comment:
do we want `process` to be available globally in all browser type files? seems like this might be a good case for one-of exceptions in individual files if we have any

Reply:
I think it's available everywhere in the renderer process though
https://www.electronjs.org/docs/latest/api/process#sandbox

Reply:
I was sure that some renderer files have a global, anyway I'll try the exception route and put what I find here

Reply:
we must be doing something to inject this value somewhere, though I'm not sure where. when I try to print `process.env` to the console it shows `undefined`, but `process.env.NODE_ENV` shows "development". are we somehow mapping this?

Reply:
think it's due to bullet point #4 here: https://en.parceljs.org/env.html#%F0%9F%8C%B3-environment-variables